l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,363 @@
+        config.eraChainId = toml.readUint("$.chain.era_chain_id");
+        config.chainAdmin = toml.readAddress("$.chain.admin");
+        config.diamondCutData = toml.readBytes("$.chain.diamond_cut_data");
+        config.bridgehubCreateNewChainSalt = toml.readUint("$.chain.bridgehub_create_new_chain_salt");
+        config.baseToken = toml.readAddress("$.chain.base_token_addr");
+        config.governance = Bridgehub(config.bridgehub).owner();
+
+        bytecodes.l2SharedBridgeBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/contracts/bridge/L2SharedBridge.sol/L2SharedBridge.json"
+        );
+
+        bytecodes.l2SharedBridgeProxyBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol/TransparentUpgradeableProxy.json"
+        );
+        bytecodes.beaconProxy = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol/BeaconProxy.json"
+        );
+    }
+
+    function checkBaseTokenAddress() internal view {
+        if (config.baseToken == address(0)) {
+            revert("Base token address is not set");
+        }
+
+        // Check if it's ethereum address
+        if (config.baseToken == ADDRESS_ONE) {
+            return;
+        }
+
+        // Does not prevent registering a non-ERC-20 contract as a token
+        // But calling the ERC-20 methods here on non-ERC-20 will fail without a readable revert message
+        if (config.baseToken.code.length == 0) {
+            revert("Token address is an EOA");
+        }
+
+        console.log("Using base token address:", config.baseToken);
+    }
+
+    function prepareRegisterBaseTokenCall() internal view returns (IGovernance.Call memory) {
+        Bridgehub bridgehub = Bridgehub(config.bridgehub);
+
+        bytes memory data = abi.encodeCall(bridgehub.addToken, (config.baseToken));
+
+        return IGovernance.Call({target: config.bridgehub, value: 0, data: data});
+    }
+
+    // @dev Computes the address of the L2 bridge and the L2 bridge proxy
+    // The variables that are used to compute the address are:
+    // - Salt
+    // - L2 bridge impl bytecode hash
+    // - Era chain ID
+    // - Beacon proxy bytecode hash
+    // - L1 governance address(owner of the L2 bridge)
+    // - L1 ERC20 bridge proxy address
+    // - Default ERC20 proxy address
+    // - Deployer's address
+    function computeL2BridgeAddress() internal view returns (address) {
+        bytes32 salt = "";
+        bytes32 bridgeBytecodeHash = L2ContractHelper.hashL2Bytecode(bytecodes.l2SharedBridgeBytecode);
+        bytes memory bridgeConstructorData = abi.encode(config.eraChainId);
+
+        address deployer;
+        address l2GovernanceAddress;
+
+        // ZKsync's protocol design assumes that the addresses of all the smart contracts that are sending L1->L2
```

Comment:
this comment is not fully correct: 
The deployer address will depend on whether you are sending the deploy request from L1 (then every address will be aliased) or whether you deploy via L2 message directly.

Reply:
are you sure that's the case for EOAs? only contracts are aliased [here](https://github.com/matter-labs/era-contracts/blob/dev/l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol#L305)

Reply:
I'll have to check, but we can do it async.